myhpi/tests/polls/test_ranked_choice_algorithm.py

```diff
@@ -0,0 +1,123 @@
+import json
+from contextlib import contextmanager
+from datetime import datetime, timedelta
+
+from django.db import connections
+from django.test.utils import CaptureQueriesContext, tag
+
+from myhpi.polls.models import (
+    PollList,
+    RankedChoiceBallot,
+    RankedChoiceBallotEntry,
+    RankedChoiceOption,
+    RankedChoicePoll,
+)
+from myhpi.tests.core.utils import MyHPIPageTestCase
+
+
+class RankedChoiceAlgorithmTests(MyHPIPageTestCase):
+    def setUp(self):
+        super().setUp()
+
+        method = getattr(self, self._testMethodName)
+        tags = getattr(method, "tags", {})
+        if "skip_setup" in tags:
+            return
+        self.setup_poll()
+
+        self.options = {
+            "alice": RankedChoiceOption.objects.create(name="Alice", poll=self.poll),
+            "bob": RankedChoiceOption.objects.create(name="Bob", poll=self.poll),
+            "charlie": RankedChoiceOption.objects.create(name="Charlie", poll=self.poll),
+            "dora": RankedChoiceOption.objects.create(name="Dora", poll=self.poll),
+        }
+
+    def setup_poll(self):
+        self.poll_list = PollList(
+            title="Polls",
+            slug="polls",
+            path="0001000200010005",
+            depth=4,
+            is_public=True,
+        )
+        self.information_menu.add_child(instance=self.poll_list)
+
+        self.poll = RankedChoicePoll(
+            title="SLASH 1999",
+            slug="slash-1999",
+            description="Who should win the SLASH 1999?",
+            start_date=datetime.now() - timedelta(days=1),
+            end_date=datetime.now() + timedelta(days=1),
+            results_visible=False,
+            is_public=True,
+        )
+
+        self.poll_list.add_child(instance=self.poll)
+
+    def cast_ballots(self, template_ballots):
+        for template_ballot in template_ballots:
+            ballot = RankedChoiceBallot.objects.create(poll=self.poll)
+            ballot.save()
+
+            for idx, template_entry in enumerate(template_ballot):
+                RankedChoiceBallotEntry.objects.create(
+                    ballot=ballot, option=self.options[template_entry], rank=idx
```

Comment:
Maybe using the POST route would be nice here? That would actually test some of the behavior there. Maybe a test would then be possible where two requests are sent at the same time, though I am actually not sure if that is possible to be implemented in a non-flaky way.